kernel/core/os.adb

```diff
@@ -449,24 +448,20 @@ is
       Global => (Input => (os_task_rw)),
       Pre => os_ghost_task_mbx_are_well_formed (task_id)
    is
-      mbx_mask  : os_mbx_mask_t;
+      mbx_mask  : os_mbx_mask_t := 0;
       mbx_index : os_mbx_index_t;
    begin
 
-      mbx_mask := 0;
-
-      if os_mbx_get_mbx_count (task_id) > os_mbx_count_t'First then
-
-         for iterator in os_mbx_count_t'First ..
-		         (os_mbx_get_mbx_count (task_id) - os_mbx_count_t'(1))
+      if not os_mbx_is_empty (task_id) then
+         mbx_index := os_mbx_get_mbx_head (task_id);
+         for iterator in 1 .. os_mbx_get_mbx_count (task_id)
 	 loop
-            mbx_index := os_mbx_get_mbx_head (task_id) + iterator;
-
-            mbx_mask :=
+             mbx_mask :=
               mbx_mask or
               os_mbx_mask_t (Shift_Left
                 (Unsigned_32'(1),
                  Natural (os_mbx_get_mbx_entry_sender (task_id, mbx_index))));
+            mbx_index := os_mbx_index_t'Succ(mbx_index);
```

Comment:
Funny, somehow this is breaking some of the proof that were OK previously. This proof tool is really fragile.

Reply:
oops I'll have a look.

Reply:
Well it might only be that these provers are very fragile